wott.py

```diff
@@ -0,0 +1,168 @@
+                pubsub_topic_name = f"projects/{project_name}/topics/{pubsub_topic}"
+            ),
+            credentials = [types.RegistryCredential(
+                public_key_certificate=types.PublicKeyCertificate(
+                    format=enums.PublicKeyCertificateFormat.X509_CERTIFICATE_PEM,
+                    certificate=ca_cert
+                )
+            )],
+            http_config = types.HttpConfig(
+                http_enabled_state=enums.HttpState.HTTP_DISABLED
+            )
+        )
+        registry = client.create_device_registry(location_path, registry)
+    return registry
+
+
+def create_or_update_device(client, project_name, location_name, registry_name, device_id, device_cert):
+    device_name = client.device_path(project_name, location_name, registry_name, 'wott-' + device_id)
+    try:
+        device = client.get_device(device_name)
+    except NotFound:
+        print(f'Creating new device {device_name}')
+        device = types.Device(
+            id='wott-' + device_id,
+            # name should be empty
+            credentials=[
+                types.DeviceCredential(
+                    public_key=types.PublicKeyCredential(
+                        format=enums.PublicKeyFormat.ES256_X509_PEM,
+                        key=device_cert
+                    )
+                )
+            ]
+        )
+        registry_path = client.registry_path(project_name, location_name, registry_name)
+        device = client.create_device(registry_path, device)
+    else:
+        print(f'Updating device {device_name}')
+        device = types.Device(
+            name = device_name,
+            credentials=[
+                types.DeviceCredential(
+                    public_key=types.PublicKeyCredential(
+                        format=enums.PublicKeyFormat.ES256_X509_PEM,
+                        key=device_cert
+                    )
+                )
+            ]
+        )
+        client.update_device(device, types.FieldMask(paths=['credentials']))
+    return device
+
+
+def get_ca_cert(debug):
+    ca = requests.get(f'{WOTT_ENDPOINT}/v0.2/ca-bundle')
+
+    if debug:
+        print("[RECEIVED] Get CA Cert: {}".format(ca.status_code))
+        print("[RECEIVED] Get CA Cert: {}".format(ca.content))
+
+    if not ca.ok:
+        print('Failed to get CA...')
+        print(ca.status_code)
+        print(ca.content)
+        return
+
+    return ca.json()['ca_bundle']
+
+
+def get_device_cert(device_id, debug):
+    cert_request = requests.get(f'{WOTT_ENDPOINT}/v0.2/device-cert/{device_id}')
+
+    if debug:
+        print("[RECEIVED] Get CA Cert: {}".format(cert_request.status_code))
+        print("[RECEIVED] Get CA Cert: {}".format(cert_request.content))
+
+    if not cert_request.ok:
+        print('Failed to get CA...')
+        print(cert_request.status_code)
+        print(cert_request.content)
+        return
+
+    return cert_request.content
+
+
+
+if __name__ == '__main__':
```

Comment:
Pls carefully check all arguments' `required` and `default` values. Many of them look wrong